commcare-cloud/commcare_cloud/environment/paths.py

```diff
@@ -17,51 +16,35 @@ def __init__(self, env_name):
 
     @memoized_property
     def public_yml(self):
-        return get_public_vars_filepath(self.env_name)
+        return os.path.join(ENVIRONMENTS_DIR, self.env_name, 'public.yml')
 
     @memoized_property
     def vault_yml(self):
-        return get_vault_vars_filepath(self.env_name)
+        return os.path.join(ENVIRONMENTS_DIR, self.env_name, 'vault.yml')
 
     @memoized_property
     def known_hosts(self):
-        return get_known_hosts_filepath(self.env_name)
+        return os.path.join(ENVIRONMENTS_DIR, self.env_name, 'known_hosts')
 
     @memoized_property
     def inventory_ini(self):
-        return get_inventory_filepath(self.env_name)
+        return os.path.join(ENVIRONMENTS_DIR, self.env_name, 'inventory.ini')
 
     @memoized_property
     def app_processes_yml(self):
-        return get_app_processes_filepath(self.env_name)
+        return os.path.join(REPO_BASE, 'environments', self.env_name, 'app-processes.yml')
 
     @memoized_property
     def app_processes_yml_default(self):
-        return get_default_app_processes_filepath()
-
-
-def get_public_vars_filepath(environment):
-    return os.path.join(ENVIRONMENTS_DIR, environment, 'public.yml')
-
-
-def get_vault_vars_filepath(environment):
-    return os.path.join(ENVIRONMENTS_DIR, environment, 'vault.yml')
-
-
-def get_known_hosts_filepath(environment):
-    return os.path.join(ENVIRONMENTS_DIR, environment, 'known_hosts')
-
-
-def get_inventory_filepath(environment):
-    return os.path.join(ENVIRONMENTS_DIR, environment, 'inventory.ini')
-
-
-def get_app_processes_filepath(env_name):
-    return os.path.join(REPO_BASE, 'environments', env_name, 'app-processes.yml')
+        return os.path.join(REPO_BASE, 'environmental-defaults', 'app-processes.yml')
 
+    @memoized_property
+    def fab_settings_yml(self):
+        return os.path.join(ENVIRONMENTS_DIR, self.env_name, 'fab-settings.yml')
 
-def get_default_app_processes_filepath():
-    return os.path.join(REPO_BASE, 'environmental-defaults', 'app-processes.yml')
+    @memoized_property
```

Comment:
do these really need to be memoized? They don't seem to be very expensive

Reply:
They definitely don't need to be memoized for performance reasons. It also doesn't _hurt_ performance, so I'm opting them to use these decorators to signal to the reader that these are lazily computed immutable properties. I like to signal things like that where I can, just to narrow down what the reader has to think about when debugging later. (I.e. they never have to consider the thought that maybe the value of `REPO_BASE` changed between calls or something.) If that's not how you read it though, maybe it's ineffective at serving that purpose.

Reply:
that makes sense but isn't the way I think about `memoized`. What about aliasing it to `lazy_immutable_property`?